### PR TITLE
All the tasks from exceptions

### PR DESCRIPTION
**Pull request by aleksandrhristov**
All the 4 tasks about exceptions. Third and Fourth tasks are in one which name is list.

### CODE REVIEW

File: src/readingintfromconsole/ConsoleReader.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package readingintfromconsole;
+
+import java.util.Scanner;
+
+
+/**
+ * Created by clouway on 03.05.16.
+ */
+public class ConsoleReader {
```
Comment (mgenov) [inline src/readingintfromconsole/ConsoleReader.java]: Doesn't sound reasonable ConsoleReader to have method checkifthenumberisintherange. 

ConsoleReader reader = new ConsoleReader(); 
reader.checkifthenumberisintherange(10,20); 

Also this method reads from console and nothing in name of the method says so.

File: src/readingintfromconsole/ConsoleReader.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package readingintfromconsole;
```
Comment (mgenov) [inline src/readingintfromconsole/ConsoleReader.java]: Package naming need to be improved: com.clouway.io.range 

Range/Boundary are good candidates for naming of this task.

File: src/readingintfromconsole/ConsoleReader.java (hunk 1 above)
Comment (mgenov) [inline src/readingintfromconsole/ConsoleReader.java]: Keep in mind that reading from console could be executed somewhere else, cause method is now doing 2 things.

File: src/sumator/Sumator.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    public String sum(String a, String b) {
+        String result = "";
+
+        try {
+            int value = Integer.parseInt(a) + Integer.parseInt(b);
+
+            result = "" + value;
+
+        } catch (NumberFormatException ex) {
```
Comment (mgenov) [inline src/sumator/Sumator.java]: This is how will look my code if I follow your logic:

```java
String result = sumator.sum("a","5");
if ("Invalid input".equals(result)) {
    // Oh, I got wrong input ! 
}
```

So I have to know what you are returning for error. If exception is thrown (checked maybe?), you will have:

```java
try {
 String result = sumator.sum("a","5"); 
} catch(GoodNamedException e) {
  // Ah I got error that my input is wrong, but I had to looked into the source of Sumator to find the magic word "Invalid input"
}
```

File: src/list/List.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+package list;
+
+
+/**
+ * Created by clouway on 03.05.16.
+ */
+public class List {
+    Object[] array;
+    int index = 0;
+    int size = 0;
```
Comment (mgenov) [inline src/list/List.java]: Why you need to keep the size ?

File: src/list/List.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    public void add(Object object) throws Exception {
+        if (index < size) {
+            array[index] = object;
+            index++;
+        } else {
+            throw new Exception("The Array is full.");
+        }
+    }
+
+    public void remove()  throws Exception {
```
Comment (mgenov) [inline src/list/List.java]: Looks like method is not formatted.

**Comment by mgenov** [comment]
You are having 2 commits and one of them is a merge commit. This doesn't look clean.

File: src/readingintfromconsole/ConsoleReader.java (hunk 1 above)
Comment (aleksandrhristov) [inline src/readingintfromconsole/ConsoleReader.java]: fixed

File: src/sumator/Sumator.java (hunk 3 above)
Comment (aleksandrhristov) [inline src/sumator/Sumator.java]: fixed

File: src/list/List.java (hunk 4 above)
Comment (aleksandrhristov) [inline src/list/List.java]: fixed

File: src/readingintfromconsole/ConsoleReader.java (hunk 2 above)
Comment (aleksandrhristov) [inline src/readingintfromconsole/ConsoleReader.java]: fixed

File: src/com/clouway/sumator/Sumator.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import java.math.BigInteger;
+
+/**
+ * Created by clouway on 14.04.16.
+ */
+public class Sumator {
+
+
+
+    public String sum(String a, String b) {
```
Comment (mgenov) [inline src/com/clouway/sumator/Sumator.java]: The sum method need to handle exceptions that could occur during the parsing of integers and re-throw them as input errors to the main.

File: src/com/clouway/sumator/SumatorDemo.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Created by clouway on 14.04.16.
+ */
+public class SumatorDemo {
+    public static void main(String[] args)   {
+        Sumator sumator = new Sumator();
+        String s = "";
+        try {
+            s = sumator.sum("jk", "5");
+        } catch (NumberFormatException e){
```
Comment (mgenov) [inline src/com/clouway/sumator/SumatorDemo.java]: Exceptions are not handled that way as they are just re-thrown.

**Comment by mgenov** [comment]
LGTM !

**Pull request merged**